scala-ts-mode.el

```diff
@@ -579,6 +607,14 @@ Return nil if there is no name or if NODE is not a defun node."
     (setq-local
      treesit-simple-indent-rules scala-ts--indent-rules)
 
+    ;; Navigation.
+    (setq-local treesit-defun-type-regexp
+                (regexp-opt '("class_definition"
+                              "object_definition"
+                              "trait_definition"
+                              "function_definition"
+                              "val_definition")))
```

Comment:
Could we use `rx` package's `or` construct here instead? I'd ideally like to keep all regex related things using that. What would the difference in the two outputs be?

Reply:
Done